src/matrix.rs

```diff
@@ -26,6 +26,48 @@ macro_rules! matrix {
                 [$( self.$field.into() ),*]
             }
         }
+
+        #[cfg(feature = "serde")]
+        impl<T> ::serde::Serialize for $name<T>
+            where $vec<T>: ::serde::Serialize
+        {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                where S: ::serde::Serializer
+            {
+                use ::serde::ser::SerializeSeq;
+                let mut seq = serializer.serialize_seq(None)?;
```

Comment:
don't we know the size here?

Reply:
Sort of, I'd need to add another variable to the macro though, since it would be somewhat difficult to extract from the fixed arrays.

Reply:
`mem::size_of::<Self>() / mem::size_of::<T>()` perhaps?

Reply:
`mem::size_of::<Self>() / mem::size_of_val(&self.x)` perhaps?